### DISCUSSION TITLE
Javascript API?

### DISCUSSION THREAD

**Discussion by lougreenwood** [zachowj/node-red-contrib-home-assistant-websocket#1052] [Feature Request]
Maybe I missed this, but did you consider exposing the implementation also as a Javascript API?

For example, there are times when calling a service inline in a function node would be useful instead of splitting a function node into 2 separate nodes with a service call in-between.

More specifically, the more complex my flows get, the more I hit against the limits of the node-red model and I want to drop further back to a JS-only implementation.

**Comment by zachowj** [comment]
From my limited knowledge of Node-RED, I don't believe it's possible for 3rd party packages to modify what is exposed in the function node.

A possible solution for you is to define your access token in the global context and use your choice of HTTP client to make a call to the Home Assistant [REST API](https://developers.home-assistant.io/docs/api/rest).
